### PR TITLE
[WIP] Created area and page to display brands using fastify

### LINKED ISSUE #73: Create a new page on fastify users
As titled!

### PR DESCRIPTION
**Pull request by lmammino**
This is currently **work in progress**.

I will be adding screenshots and more details to start receiving feedbacks.

Closes #73 

**PREVIEW**:

- https://build-jyezjymwok.now.sh/
- https://build-jyezjymwok.now.sh/organisations

Questions:

 - How do we get the first 6 brands? Do we have enough among the contributors? If not, can we shout out (maybe on twitter) that we are collecting companies using fastify?
 - I am not sure about the naming here: I ~~am~~ was using a mixture of "brand", "company", "organisation". Maybe it's good to pick only one and be consistent. What do you think? (Now i am using **organization/organisations** as it feels the most comprehensive/generic to me).
 - How do we decide which brands stay on the home page? Do we sort/feature them manually? Do we use an objective metric (followers on twitter / team size / company size / company revenues / etc.)? If so, which one?

---

Todo:

- [x] Decide how to select the logos in the home page (for now taking only the first 6 in alphabetical order)
- [x] Review and agree on the content and copy check
- [ ] Collect the first 6 **REAL** brands

### CODE REVIEW

**Comment by lmammino** [comment]
Current design in home page.

Logos are fictional for now and **fast banana** is obviously our best user! 🙄 

screen shot 2018-09-08 at 15 56 00

---

screen shot 2018-09-08 at 15 56 17

---

screen shot 2018-09-08 at 15 56 36

**Comment by lmammino** [comment]
The idea for the brands page:

screen shot 2018-09-08 at 17 03 57

**Comment by lmammino** [comment]
Full text on how to contribute and add your own logo:

---

<p>Is your company using fastify? That's great!</p>
          <p>If you would like your company to be featured in this list, we would love it too!</p>
          <p>Here's how to submit your company, your logo and other few relevant details, so that we can update this page:</p>
          <ol>
            <li>Fork the <a href="https://github.com/fastify/website">Fastify website repository</a> on GitHub.</li>
            <li>
              Add the logo in the folder <code>src/website/content/images/brands</code>.<br/>
              Please make sure that your logo complies to the following specs:
              <ul>
                <li>Encoded in SVG (Scalable Vector Graphics) format</li>
                <li>Use only 2 colors: <code>#fff</code> (white) and <code>#999</code> (gray)</li>
                <li>Be reasonably optimized/minified (you can use <a href="https://www.svgminify.com">svgminify</a>)</li>
              </ul>
            </li>
            <li>Update the file <code>src/website/data/brands.yml</code> with a new entry (in alphabetical order by company name).<br/>
              Your entry must provide:
              <ul>
                <li><code>name</code>: the name of your company</li>
                <li><code>image</code>: the filename of the file you added in the previous step</li>
                <li><code>link</code>: a link to a website that represents your organisation</li>
              </ul>
              Please, make sure not to break the YAML syntax of the file.
            </li>
            <li>Submit a PR!</li>
          </ol>
          <p>As soon as your PR is reviewed and merged, your awesome logo will appear in this page.</p>

**Comment by mcollina** [comment]
I don’t think it’s feasible to use just those two colors. Companies should be able to use their full branding.


You can also add a shortcut.. send an svg to <email>.

**Comment by lmammino** [comment]
I kind of agree on your point about colors @mcollina. But I wanted to keep the logos in grayscale to avoid to clutter the design, at least in the home page.

I just figured out a way to have colored images and display them as grayscale when needed through CSS filters ([demo here](https://codepen.io/lmammino/pen/gdopvR)). I am going to update this PR and the PR description accordingly.

Regarding:

> You can also add a shortcut.. send an SVG to .

I am not sure what you mean? Are you suggesting that we should be able to receive SVGs in some other way and do the PR by ourselves?

Also, do you have some feedback on my other questions in the PR description?

**Comment by lmammino** [comment]
I updated the PR with the following changes:

- Using only "organization/organisations" as a way to identify companies/brands/users adopting fastiy (it feels the most comprehensive to me, but feel free to suggest otherwise).
- Removed the note about grayscale logos, now using fully coloured logos and displaying them using the CSS grayscale filter (that is removed on mouse over). This seems to guarantee a clean enough design without penalizing too much the brand identity of the organisations.
- Also reviewed the copy a bit, but please double check that everything feels clear and friendly enough.

---

See a live demo of the current version here: https://build-ojwnkkkuyj.now.sh/

---

Looking forward to having your feedback 😉 Have a happy Sunday! 🌞

**Comment by mcollina** [comment]
I would prefer if we just visualized them all for now, and start with what we have. Currently, Vectra, NearForm, Knock, LetzDoIt (cc @lucamaraschi), YNAP (@thearegee).

If possible, I would randomize display order.

**Comment by lmammino** [comment]
Randomizing is actually a great idea. Every build will be likely to feature a different set of companies (in the home page) and a randomly ordered list in the page containing all the organisations. Sounds very fair. I'll apply this change shortly.

**Regarding having the first logos, as soon as you have your bosses approval can you guys upload here the SVG?** I'll take care of doing everything else.

CC:

 - Nearform (@mcollina)
 - Knock (@jsumners)
 - LetzDoIt (@lucamaraschi)
 - YNAP (@thearegee)
 - MIA platform (@allevo)

Any chance to get your own bosses big names @evanshortiss and @delvedor? 😛

@nwoltman, do you want to feature your startup too (if you are using fastify)?

@trivikr & @StarpTech do you have any organisation you might want to feature?

**Comment by lmammino** [comment]
New preview(s):

- https://build-jyezjymwok.now.sh/
- https://build-jyezjymwok.now.sh/organisations

**Comment by robinglen** [comment]
I was thinking of only using Mr Porter - as for now thats the only website that its deployed on. We will be looking to include NET-A-PORTER and The Outnet later this year.

What are thoughts on this?

Do you want us to add the logos to this branch `feature/brands-using-fastify`?

**Comment by StarpTech** [comment]
@lmammino great work! Currently, I don't use fastify inside an organization but the future looks bright 😄 

@lmammino The urls of the logos aren't done right?

**Comment by lmammino** [comment]
@thearegee, for now, it's probably just easier to upload the logo here as a comment (attachment). 

@StarpTech 

> The urls of the logos aren't done right?

Not sure what you mean here

**Comment by jsumners** [comment]
I like the developed plan thus far. I see about getting the SVG tomorrow.

**Comment by StarpTech** [comment]
@lmammino some doesn't support https, many points to example.con

**Comment by robinglen** [comment]
I've added it as a .zip because github didn't like .svg.

As mentioned above this for Mr Porter as this is the only brand we have deployed on so far. Url is:
[https://www.mrporter.com](https://www.mrporter.com)

[mrp.svg.zip](https://github.com/fastify/website/files/2364258/mrp.svg.zip)

**Comment by lmammino** [comment]
@thearegee brilliant. thanks :)

@StarpTech those are just (funny) fictional examples, even though i would love to buy from fast banana 🤣 i will replace them as soon as we have few real organisations.

**Comment by StarpTech** [comment]
@lmammino haha 😂 I wouldn't be surprised when it exists.. Fast banana blockchain.

**Comment by nwoltman** [comment]
@lmammino It's a little too early to have my startup featured anywhere. I haven't even released a product yet 😅

**Comment by lmammino** [comment]
@nwoltman, just don't forget about us when your startup will be super successful and you'll be a millionaire 😛 

Got official approval from Vectra too 🎉. Gonna start to upload some real logos.

So that's the current state:

 - [X] Mr Porter
 - [X] Vectra 
 - [x] Knock (@jsumners)
 - [x] MIA platform (@allevo)
 - [ ] LetzDoIt (@lucamaraschi)
 - [x] Nearform (@mcollina)

I think as soon as we have those initial ones we can go live and then add new ones as we go.

**Comment by mcollina** [comment]
We should add a notice somewhere in the folder with the logos that they are property of the respective companies, and are not covered under MIT. This is kind of obvious, but better be safe.

**Comment by lmammino** [comment]
Very good idea @mcollina. I'll add it to the page and also add a txt in the folder.

**Comment by jsumners** [comment]
@lmammino here's Knock's (knock.com).

[knock-logo.svg.zip](https://github.com/fastify/website/files/2366971/knock-logo.svg.zip)

**Comment by allevo** [comment]
https://www.mia-platform.eu/

[mia-platform.zip](https://github.com/fastify/website/files/2367055/mia-platform.zip)

@lmammino can we add some sentencies ? for example "About us" / "hear from" / "press" like https://hapijs.com/community ??

**Comment by lmammino** [comment]
Thanks @jsumners and @allevo, I'll be adding them tonight.

@allevo, regarding the "quotes/phrases", that would be very cool, but it would definitely be extra work at the moment, so I would suggest exploring that option later, maybe with a more specific section called "case studies".

**Comment by jsumners** [comment]
Agreed. It would also make it more difficult to submit a logo.

**Comment by lmammino** [comment]
Updated the website with Mia Platform and Knock! 🎉

Here you have an updated preview

---

https://build-nwujmuprqa.now.sh/

https://build-nwujmuprqa.now.sh/organisations

---

@mcollina et al. I also added a disclaimer about logo ownership as you suggested. Please let me know if my improvised "legalese" language seems good enough!

Looking forward for the last 2 logos. 🤟

**Comment by evanshortiss** [comment]
@lmammino I'm not aware of anything in production yet, but i will PR as soon as I learn of something!

**Comment by lmammino** [comment]
@mcollina @lucamaraschi, any update? is there any way I can get any vector format for the logos? I can take care of converting them into SVG if needed.

**Comment by jsumners** [comment]
@lmammino you have instances of "fastify" that need to be "Fastify" in the text.

**Comment by lmammino** [comment]
Good catch @jsumners, gonna update soon

**Review by jsumners — APPROVED** [review]
LGTM

**Review by mcollina — APPROVED** [review]
LGTM

**Review by allevo — APPROVED** [review]
LGTM

**Comment by lmammino** [comment]
Gonna get this one merged even without the logo from letzdoit that we can add it later. @lucamaraschi, give us a shout when you are ready (or feel free to open a PR) 😉 

I think we should tweet something from the official Fastify account to mention that we are starting to collect companies using Fasitfy :)

**Comment by delvedor** [comment]
I'm sorry that I wasn't able to follow this more closely, I was traveling for work, and I had limited time.
Great work!! :)

File: src/website/content/images/organisations/LICENSE [hunk 1]
```diff
@@ -0,0 +1,2 @@
+The logos provided here are property of the respective organisations and they are
+not distributed under the same license as fastify (MIT).
```
Comment (nwoltman) [inline src/website/content/images/organisations/LICENSE:2]: Fastify

File: src/website/content/organisations.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+---
+title: Orgaisations using fastify
```
Comment (nwoltman) [inline src/website/content/organisations.md:2]: Typo. Should be: `Organisations using Fastify`

**Comment by nwoltman** [comment]
Whoops, didn't realize this was already merged 😅

**Comment by robinglen** [comment]
@lmammino @mcollina both [the outnet](https://www.theoutnet.com/en-gb/) and [net-a-porter](https://www.net-a-porter.com/gb/en/) are now using fastify too. Thats a few extra hundred million of page views per month, if you were interested in the scale of traffic.

If you wanted to add them to your docs too.

**Comment by mcollina** [comment]
It would be fantastic! Would you like to send a PR?

**Comment by lmammino** [comment]
Wonderful news @thearegee! It would be great to have both logos added to [our collection](https://www.fastify.io/organisations/)

**Comment by robinglen** [comment]
@mcollina + @lmammino I will happy send a PR but as we previously just added it all into here, just let me know the preferred way you would like to raise it and I will get you the logos etc.

Thanks,
Robin

**Comment by mcollina** [comment]
If you don’t mind to send a PR that edits: https://github.com/fastify/website/blob/master/src/website/data/organisations.yml

**Comment by robinglen** [comment]
on my todo for tomorrow @mcollina :)

**Pull request merged**